rosidl_generator_cpp/test/test_array_generator.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_ARRAY_GENERATOR_HPP_
+#define TEST_ARRAY_GENERATOR_HPP_
+
+#include <climits>
+#include <random>
+
+/**
+ * Helper function to generate a test pattern for boolean type.
+ * Alternating true (even index) and false (odd index) pattern.
+ */
+template<typename C>
+void test_vector_bool(C & container, int size)
```

Comment:
Consider using an unsigned type (like `size_t`) instead of `int` for this argument and loop iterations.